### PR TITLE
Add support for entering tax as a percentage of Total

### PR DESCRIPTION
**Pull request by arimbun**
Simple support to configure tax as a percentage of Total in the `billing.yaml` file.

### CODE REVIEW

**Comment by relistan** [comment]
Sorry I missed this when you opened it! Will review tomorrow. Thanks for the contribution @arimbun

**Comment by arimbun** [comment]
@relistan No problem, did you have a chance to review it?

File: billing.example.yaml [hunk 1]
```diff
@@ -33,6 +33,10 @@ billables:
     unit_price:  10.23
     currency:    €
 
+# You may enter tax as a percentage here (from 0 to 1.0).
+tax:
+  percentage: 0.1
```
Comment (relistan) [inline billing.example.yaml:38]: We really need to support a default tax and an overridable tax rate per item. Some items may be taxable at different rates.

File: billing.example.yaml [hunk 2]
```diff
@@ -33,6 +33,10 @@ billables:
     unit_price:  10.23
     currency:    €
 
+# You may enter tax as a percentage here (from 0 to 1.0).
+tax:
```
Comment (relistan) [inline billing.example.yaml:37]:
```suggestion
default_tax:
```

File: invoice/config.go [hunk 3]
```diff
@@ -65,6 +65,10 @@ func (b *BillableItem) Strings() []string {
 	}
 }
 
+type TaxDetails struct {
```
Comment (relistan) [inline invoice/config.go:68]: We should call this something like `DefaultTax` instead.

File: invoice/bill.go [hunk 4]
```diff
@@ -272,18 +276,19 @@ func (b *Bill) drawBillablesTable(headers []string, billables []BillableItem, wi
 	// Draw Tax
 	b.pdf.Ln(4)
 	b.drawBlanks(billables, widths)
-	b.textFormat(widths[len(widths)-2], 4, "Tax", "1", 0, "R", true, 0, "")
-	b.textFormat(widths[len(widths)-1], 4, "0", "1", 0, "R", true, 0, "")
+	taxText := billables[0].Currency + " " + niceFloatStr(tax)
+	b.textFormat(widths[len(widths)-2], 4, "GST", "1", 0, "R", true, 0, "")
```
Comment (relistan) [inline invoice/bill.go:280]:
```suggestion
	b.textFormat(widths[len(widths)-2], 4, "Tax", "1", 0, "R", true, 0, "")
```

GST is a region-specific tax for certain countries. Let's stay with generic "Tax"

**Review by relistan — CHANGES_REQUESTED** [review]
@arimbun MANY apologies for taking so darn long to get to this. It's a side project and finding time can be a challenge. This looks pretty good. I would like to eventually allow item-specific tax calculation, so I've pointed out where I'd like to change what you've got here. Otherwise, good to go. Thank you for the contribution! Again, sorry about how long it took to get to it. @rolandtritsch 's PR was waiting on me for quite awhile as well.

File: billing.example.yaml (hunk 2 above)
Comment (relistan) [reply billing.example.yaml:37]: You'll need to make various adjustments elsewhere in the code to make this work also. Didn't try to find them all.

File: invoice/bill.go (hunk 4 above)
Comment (relistan) [reply invoice/bill.go:280]: Alternatively, make this configurable in the `default_tax` block in the config.

**Comment by relistan** [comment]
Hey @arimbun I've got a little time to deal with this today so I'm going to merge it as-is and do a little work on it now. Thanks again for the contribution!

**Comment by rolandtritsch** [comment]
@relistan @arimbun Nice!!!

**Pull request merged**